### PR TITLE
Task/dr 432 replicate cm ia ccode

### PR DESCRIPTION
**Pull request by Pira-nhs**
## Description

Replicate the infrastructure code from uec-cm to uec-dos-service-management.
References to 'cm' has been updated of 'sm'

## Context

Change is required to have the backend infrastructure deployed in AWS for the DOS Dev environment.

## Type of changes

- [ ] Refactoring (non-breaking change)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [x] I have followed the code style of the project
- [ ] I have added tests to cover my changes
- [ ] I have updated the documentation accordingly
- [x] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [x] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

File: infrastructure/stacks/application/scripts/database_populator/database_populator.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+import boto3
+from io import BytesIO
+
+# Initialize the S3 and DynamoDB clients
+s3 = boto3.client("s3")
+dynamodb = boto3.resource("dynamodb")
+ddbclient = boto3.client("dynamodb")
+
+# Specify the S3 bucket name and Excel file key
+bucket_name = "nhse-uec-cm-dev-databucket"
```
Comment (tamaragoldschmidt) [inline infrastructure/stacks/application/scripts/database_populator/database_populator.py:14]: References here to cm

Comment (Pira-nhs) [reply infrastructure/stacks/application/scripts/database_populator/database_populator.py:14]: Nachi is working on updating this script. When that's done will bring in the updated script with reference to 'sm'

File: .gitignore [hunk 2]
```diff
@@ -1,5 +1,54 @@
+<<<<<<< HEAD
```
Comment (tamaragoldschmidt) [inline .gitignore:1]: Has this come in from a rebase?

File: .gitignore [hunk 3]
```diff
@@ -1,5 +1,54 @@
...
+htmlcov/
+*/test
+*/*.pyc
+__pycache__
+*/__pycache__
+*.pyc
+.coverage.*
+coverage.xml
+tests/*/reports/*
+>>>>>>> d5b0766 (DR-432 Include support tools)
```
Comment (jope1) [inline .gitignore:43]: And this (line 43)

File: .coveragerc [hunk 4]
```diff
@@ -0,0 +1,19 @@
+
+[run]
+branch = True
+omit =
+        */test/*
+        */usr/local/lib*
+        */__init__.py
+        */*/chalicelib/common/utilities.py
+        */app.py
+        */database_populator.py
```
Comment (jope1) [inline .coveragerc:10]: suspect this would need to be **/*database_populator.py too.

File: .coveragerc [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+[report]
+omit =
+        */test/*
+        */usr/local/lib*
+        */__init__.py
+        */*/chalicelib/common/utilities.py
+        */app.py
+        */database_populator.py
```
Comment (jope1) [inline .coveragerc:19]: suspect this would need to be **/*database_populator.py too.

**Review by jope1 — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 2 above)
Comment (Pira-nhs) [reply .gitignore]: correct good catch! sorted now

File: .coveragerc (hunk 4 above)
Comment (Pira-nhs) [reply .coveragerc]: agreed..have changed it

**Review by jope1 — APPROVED** [review]

**Review by tamaragoldschmidt — APPROVED** [review]

**Pull request merged**